vars/eos/4.24.6M/vars.yml

```diff
@@ -22,15 +22,15 @@ expect_responses:
         cat << EOF > /mnt/flash/dhcp-ma1-on-boot.sh
         #!/bin/bash
 
-        dhclient ma1
+        sudo dhclient ma1
         ADDRESS=$(ifconfig ma1|grep inet|awk -F " " '{print $2}')
         NETMASK=$(ifconfig ma1|grep inet|awk -F " " '{print $4}')
         DEFAULT_GW=$(ip route|grep default|awk -F " " '{print $3}')
         FastCli -p 15 -c $a"
         configure terminal
         interface Management1
-        ip address $ADDRESS $NETMASK
-        ip route 0.0.0.0 0.0.0.0 $DEFAULT_GW
+        ip address $(ifconfig ma1|grep inet|awk -F " " '{print $2}') $(ifconfig ma1|grep inet|awk -F " " '{print $4}')
+        ip route 0.0.0.0 0.0.0.0 $(ip route|grep default|awk -F " " '{print $3}')
```

Comment:
why did we replace the variables from above? We should move this logic to lines 26 to 28

Reply:
The variables are getting substituted properly, hence removed the variables. I will remove the variable declarations.

Reply:
if the variables are substituted properly, then why doesn't the commands work as expected?

It looks like you've just move the logic from lines above to here, which is a noop.

Reply:
Sorry for the typo. The variables are not getting substituted properly. After I  used the value directly, instead of the variable, the commands are formed correctly.

Reply:
Lets do this, can you revert this change. Set
```
/bin/bash -ex
```

Then paste the output of the script. I cannot see why the variables wouldn't work properly